Fog.js

```diff
@@ -500,9 +500,10 @@ function do_check_token_visibility() {
 		let playerTokenId = $(`.token[data-id*='${window.PLAYER_ID}']`).attr("data-id");
 		let playerTokenHasVision = (playerTokenId == undefined) ? true : window.TOKEN_OBJECTS[playerTokenId].options.auraislight;
 
-		pixeldata2 = pixeldata2.filter(function(color, index) {return (index + 1) % 4 != 0});
+		//Combining some and filter cut down about 140ms for average sized picture
+		let someFilter = function(color, index) {if ((index + 1) % 4 != 0 && color == 255) return true;};

```

Comment:
we had the same idea here lol. I think we might be able to improve this more using just the edges of the token but I'm not sure as I think we'd need 4 get image datas for that a single line of pixels for each side of the square.

I had done index % 4 == 0 instead to just check the red of rgba don't think it saves much if any time though since it's still looping through them. This checks rgb and leaves out a which is fine.

I'll close my PR since you've got it here. 👍

Reply:
Nice! Great minds think alike lol. Ill make that change. Also, I almost have a web worker implementation working for the fog redraw, was thinking see web worker performance then eventually move a lot over to service worker. Little bit easier getting web worker up and running. Ill open a pr for that tomorrow probably, gonna make these changes tonight and push up